src/libraries/System.Threading.Channels/src/System/Threading/Channels/BoundedChannel.cs

```diff
@@ -16,7 +16,7 @@ internal sealed class BoundedChannel<T> : Channel<T>, IDebugEnumerable<T>
         /// <summary>The mode used when the channel hits its bound.</summary>
         private readonly BoundedChannelFullMode _mode;
         /// <summary>Task signaled when the channel has completed.</summary>
-        private readonly TaskCompletionSource<VoidResult> _completion;
+        private readonly TaskCompletionSource _completion;
```

Comment:
These changes won't work in the netstandard build of the library, where there is no non-generic TCS.  You might try creating a TCS in that build, e.g. add a file that's only compiled into that build and that provides a polyfill:
```C#
internal sealed class TaskCompletionSource : TaskCompletionSource<VoidResult> { ... }
```

Reply:
Yeah haven't thought about the netstandard part. Do you think it is worth doing so or would you as a maintainer prefer to stay with the current `TaskCompletionSource<VoidResult>`? I have no problems just closing this or do what you suggested to add a polyfill

Reply:
It's not worth complicating the code with ifdefs are partial files or anything like that.  But if such a polyfill addresses it, that'd be file.

Reply:
Pushed. I hope I got it right. Sorry this is the first time I contribute here

Reply:
Thanks!